### PR TITLE
Add `auth-helper` crate

### PR DESCRIPTION
**Pull request by Adam-Gleave**
Adds the `auth-helper` crate.

### CODE REVIEW

File: .cargo/audit.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[advisories]
```
Comment (thibault-martinez) [inline .cargo/audit.toml:1]: Can you do as in the dev branch and put this in the workflows folder ?

Comment (Adam-Gleave) [reply .cargo/audit.toml:1]: Sure :+1:

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Need to add some authorised license

**Comment by Adam-Gleave** [comment]
> Need to add some authorised license

Hopefully that should work...

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**